File: src/1503.py
class Solution:
    def getLastMoment(self, n: int, left: list[int], right: list[int]) -> int:
        if not left:
            return n - min(right)
        elif not right:
            return max(left)
        else:
            return max(max(left), n - min(right))

File: src/1503.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        if(left.empty()) {
            return n - *min_element(right.begin(), right.end());
        }
        else if(right.empty()) {
            return *max_element(left.begin(), left.end());
        }
        else {
            return max(*max_element(left.begin(), left.end()), 
            n - *min_element(right.begin(), right.end()));
        }
    }
};
